### PR TITLE
Preparesequence ignores empty tables

### PR DESCRIPTION
**Pull request by Goodmain**
I finished with the task #63.

Also I made the following:
- increased code coverage
- changed test failing to the throwing exception for the `FAIL_EXPORT_JSON` checking case
- default DB dump-file name moved to the trait's variable

### CODE REVIEW

File: src/Traits/FixturesTrait.php [hunk 1]
```diff
@@ -188,8 +190,8 @@ protected function getTables(): array
 
     protected function exportContent($content, string $fixture): void
     {
-        if (env('FAIL_EXPORT_JSON', true)) {
-            $this->fail(preg_replace('/[ ]+/mu', ' ',
+        if (config('FAIL_EXPORT_JSON', true)) {
+            throw new UnexpectedExportException(preg_replace('/[ ]+/mu', ' ',
                 'Looks like you forget to remove exportJson. If it is your local environment add
```
Comment (DenTray) [inline src/Traits/FixturesTrait.php:195]: let's move message to the exception class

File: src/Exceptions/UnexpectedExportException.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace RonasIT\Support\Exceptions;
+
+class UnexpectedExportException extends EntityCreateException
```
Comment (DenTray) [inline src/Exceptions/UnexpectedExportException.php:5]:
```suggestion
class ForbiddenExportModeException extends EntityCreateException
```

File: tests/FixturesTraitTest.php [hunk 3]
```diff
@@ -43,21 +73,96 @@ public function testGetFixtureNotExistsWithException()
...
-    public function testGetClearPsqlDatabaseQuery()
+    public function testLoadEmptyTestDump()
+    {
+        $connection = $this->mockClass(Connection::class, [], true);
+
+        $this->app->instance('db.connection', $connection);
+        $this->dumpFileName = 'clear_database/empty_dump.sql';
+
+        $connection->expects($this->never())
+            ->method('unprepared');
```
Comment (DenTray) [inline tests/FixturesTraitTest.php:84]:
```suggestion
        $connection
            ->expects($this->never())
            ->method('unprepared');
```

File: src/Traits/FixturesTrait.php [hunk 4]
```diff
@@ -188,8 +190,8 @@ protected function getTables(): array
 
     protected function exportContent($content, string $fixture): void
     {
-        if (env('FAIL_EXPORT_JSON', true)) {
-            $this->fail(preg_replace('/[ ]+/mu', ' ',
+        if (config('FAIL_EXPORT_JSON', true)) {
```
Comment (DenTray) [inline src/Traits/FixturesTrait.php:193]: let's implement package config file with this value

File: tests/FixturesTraitTest.php [hunk 5]
```diff
@@ -43,21 +73,96 @@ public function testGetFixtureNotExistsWithException()
...
+
+        $this->app->instance('db.connection', $connection);
+        $this->dumpFileName = 'clear_database/dump.sql';
+
+        Config::set('database.default', 'mysql');
+
+        self::$tables = $this->getJsonFixture('clear_database/tables.json');
+
+        $connection->expects($this->exactly(2))
+            ->method('unprepared')
```
Comment (DenTray) [inline tests/FixturesTraitTest.php:101]:
```suggestion
        $connection
            ->expects($this->exactly(2))
            ->method('unprepared')
```

File: tests/FixturesTraitTest.php [hunk 6]
```diff
@@ -43,21 +73,96 @@ public function testGetFixtureNotExistsWithException()
...
+            ->willReturn($this->getJsonFixture('prepare_sequences/tables.json'));
+
+        $connection->expects($this->once())
+            ->method('getDoctrineSchemaManager')
+            ->willReturn($mock);
+
+        $connection->expects($this->once())
+            ->method('unprepared')
+            ->with($this->getFixture('prepare_sequences/sequences.sql'))
+            ->willReturn(true);
```
Comment (DenTray) [inline tests/FixturesTraitTest.php:149]:
```suggestion
        $mock
            ->expects($this->once())
            ->method('listTableNames')
            ->willReturn($this->getJsonFixture('prepare_sequences/tables.json'));

        $connection
            ->expects($this->once())
            ->method('getDoctrineSchemaManager')
            ->willReturn($mock);

        $connection
            ->expects($this->once())
            ->method('unprepared')
            ->with($this->getFixture('prepare_sequences/sequences.sql'))
            ->willReturn(true);
```

File: tests/fixtures/FixturesTraitTest/clear_database/dump.sql [hunk 7]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO users(id, name, email, password, remember_token, set_password_hash, role_id, created_at, updated_at, set_password_hash_created_at) VALUES
+(1, 'Gerhard Feest', 'fidel.kutch@example.com', '$2y$10$X4receiTrF24bXrEbAiChOZ8TMNPqoXuhuThgynvBdWIHZeu5HzsS', null, 'good_token1', 1, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 10:11:12'),
+(2, 'Alien West', 'alien.west@example.com', 'old_password', null, 'restore_token', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 10:11:10'),
+(3, 'Alien East', 'alien.ease@example.com', 'old_password', null, 'good_token2', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 10:35:00'),
+(4, 'Alien North', 'alien.north@example.com', 'old_password', null, 'bad_token1', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 8:06:00'),
+(5, 'Alien South', 'alien.south@example.com', 'old_password', null, 'bad_token2', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 9:04:00');
```
Comment (DenTray) [inline tests/fixtures/FixturesTraitTest/clear_database/dump.sql:6]:
```suggestion
INSERT INTO users(id, name, email, password, remember_token, set_password_hash, role_id, created_at, updated_at, set_password_hash_created_at) VALUES
  (1, 'Gerhard Feest', 'fidel.kutch@example.com', '$2y$10$X4receiTrF24bXrEbAiChOZ8TMNPqoXuhuThgynvBdWIHZeu5HzsS', null, 'good_token1', 1, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 10:11:12'),
  (2, 'Alien West', 'alien.west@example.com', 'old_password', null, 'restore_token', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 10:11:10'),
  (3, 'Alien East', 'alien.ease@example.com', 'old_password', null, 'good_token2', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 10:35:00'),
  (4, 'Alien North', 'alien.north@example.com', 'old_password', null, 'bad_token1', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 8:06:00'),
  (5, 'Alien South', 'alien.south@example.com', 'old_password', null, 'bad_token2', 2, '2016-10-20 11:05:00', '2016-10-20 11:05:00', '2018-11-11 9:04:00');
```

File: tests/FixturesTraitTest.php [hunk 8]
```diff
@@ -43,21 +73,96 @@ public function testGetFixtureNotExistsWithException()
...
+        $this->dumpFileName = 'clear_database/dump.sql';
+
+        Config::set('database.default', 'pgsql');
 
-        $result = $this->getClearPsqlDatabaseQuery($tables);
+        self::$tables = $this->getJsonFixture('clear_database/tables.json');
 
-        $this->assertEquals($this->getFixture('clear_database/clear_psql_db_query.sql'), $result);
+        $connection->expects($this->exactly(2))
+            ->method('unprepared')
```
Comment (DenTray) [inline tests/FixturesTraitTest.php:122]:
```suggestion
        $connection
            ->expects($this->exactly(2))
            ->method('unprepared')
```

File: tests/FixturesTraitTest.php [hunk 9]
```diff
@@ -43,21 +73,96 @@ public function testGetFixtureNotExistsWithException()
...
+
+        $this->app->instance('db.connection', $connection);
+
+        $mock->expects($this->once())
+            ->method('listTableNames')
+            ->willReturn($this->getJsonFixture('prepare_sequences/tables.json'));
+
+        $connection->expects($this->once())
+            ->method('getDoctrineSchemaManager')
+            ->willReturn($mock);
```
Comment (DenTray) [inline tests/FixturesTraitTest.php:144]:
```suggestion
        $mock
            ->expects($this->once())
            ->method('listTableNames')
            ->willReturn($this->getJsonFixture('prepare_sequences/tables.json'));

        $connection
            ->expects($this->once())
            ->method('getDoctrineSchemaManager')
            ->willReturn($mock);
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/Traits/FixturesTrait.php (hunk 1 above)
Comment (Goodmain) [reply src/Traits/FixturesTrait.php]: @DenTray Done

File: src/Exceptions/UnexpectedExportException.php (hunk 2 above)
Comment (Goodmain) [reply src/Exceptions/UnexpectedExportException.php]: @DenTray Done

File: tests/FixturesTraitTest.php (hunk 3 above)
Comment (Goodmain) [reply tests/FixturesTraitTest.php]: @DenTray Done

File: src/Traits/FixturesTrait.php (hunk 4 above)
Comment (Goodmain) [reply src/Traits/FixturesTrait.php]: @DenTray Reverted to the env

File: tests/FixturesTraitTest.php (hunk 5 above)
Comment (Goodmain) [reply tests/FixturesTraitTest.php]: @DenTray Done

File: tests/FixturesTraitTest.php (hunk 6 above)
Comment (Goodmain) [reply tests/FixturesTraitTest.php]: @DenTray Done

File: tests/fixtures/FixturesTraitTest/clear_database/dump.sql (hunk 7 above)
Comment (Goodmain) [reply tests/fixtures/FixturesTraitTest/clear_database/dump.sql]: @DenTray Done

File: tests/FixturesTraitTest.php (hunk 8 above)
Comment (Goodmain) [reply tests/FixturesTraitTest.php]: @DenTray Done

File: tests/FixturesTraitTest.php (hunk 9 above)
Comment (Goodmain) [reply tests/FixturesTraitTest.php]: @DenTray Done

File: tests/FixturesTraitTest.php [hunk 10]
```diff
@@ -27,6 +43,30 @@ public function testGetFixture(string $input)
...
+
+        $result = [
+            'value' => 1234567890
+        ];
+
+        $this->exportJson('export_json/response.json', new TestResponse(
+            new Response(json_encode($result))
+        ));
+
+        $this->assertEquals($this->getJsonFixture('export_json/response.json'), $result);
```
Comment (DenTray) [inline tests/FixturesTraitTest.php:67]: let's assert that file was created

File: src/Exceptions/ForbiddenExportModeException.php [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+class ForbiddenExportModeException extends EntityCreateException
+{
+    public function __construct()
+    {
+        parent::__construct(preg_replace('/[ ]+/mu', ' ',
+            'Looks like you forget to remove exportJson. If it is your local environment add 
+                FAIL_EXPORT_JSON=false to .env.testing.
+                If it is dev.testing environment then remove it.'
+        ));
```
Comment (DenTray) [inline src/Exceptions/ForbiddenExportModeException.php:13]:
```suggestion
        $envKey = 'FAIL_EXPORT_JSON';
        $envFile = '.env.testing';

        parent::__construct("Looks like you try to export fixture.\nIf you see this message while running tests in the local environment - please set {$envKey}=false to {$envFile}.');
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/Exceptions/ForbiddenExportModeException.php (hunk 11 above)
Comment (Goodmain) [reply src/Exceptions/ForbiddenExportModeException.php:13]: Done

File: tests/FixturesTraitTest.php (hunk 10 above)
Comment (Goodmain) [reply tests/FixturesTraitTest.php:67]: Done

**Review by DenTray — APPROVED** [review]

**Pull request merged**